apps/css-workshop/src/components/Checkbox.astro

```diff
@@ -0,0 +1,52 @@
+---
+import cx from 'classnames';
+
+type Props = {
+  isLoading?: boolean;
+  label?: React.ReactNode;
+  indeterminate?: boolean;
+  checked?: boolean;
+  disabled?: boolean;
+  variant?: 'eyeball';
+  content?: React.ReactNode;
+  status?: 'negative' | 'positive' | 'warning' | 'negative';
+} & astroHTML.JSX.HTMLAttributes;
+
+const { status, isLoading, label, indeterminate, checked, disabled, variant, ...props } =
+  Astro.props;
+---
+
+<label
+  class={cx('iui-checkbox-wrapper', {
+    'iui-disabled': disabled,
+    'iui-loading': isLoading,
+    'iui-positive': status === 'positive',
+    'iui-warning': status === 'warning',
+    'iui-negative': status === 'negative',
+  })}
+  {...props}
+>
+  <input
+    class={cx('iui-checkbox', {
+      'iui-loading': isLoading,
+      'iui-checkbox-visibility': variant === 'eyeball',
+    })}
+    type='checkbox'
+    disabled={disabled}
+    checked={checked}
+    name={indeterminate ? 'indeterminate' : ''}
```

Comment:
Should the else part be `undefined` to prevent adding the `name` attribute when inderminate is `false`/`undefined`?

```suggestion
    name={indeterminate ? 'indeterminate' : undefined}
```